src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -124,25 +125,33 @@ private ExecutionContext CallContractInternal(ContractState contract, ContractMe
             return context_new;
         }
 
-        internal void CallFromNativeContract(UInt160 callingScriptHash, UInt160 hash, string method, params StackItem[] args)
+        internal ContractTask CallFromNativeContract(UInt160 callingScriptHash, UInt160 hash, string method, params StackItem[] args)
         {
-            ExecutionContext context_current = CurrentContext;
             ExecutionContext context_new = CallContractInternal(hash, method, CallFlags.All, false, args);
             ExecutionContextState state = context_new.GetState<ExecutionContextState>();
             state.CallingScriptHash = callingScriptHash;
-            while (CurrentContext != context_current)
-                StepOut();
+            ContractTask task = new ContractTask();
+            contractTasks.Add(context_new, task);
+            return task;
         }
 
-        internal T CallFromNativeContract<T>(UInt160 callingScriptHash, UInt160 hash, string method, params StackItem[] args)
+        internal ContractTask<T> CallFromNativeContract<T>(UInt160 callingScriptHash, UInt160 hash, string method, params StackItem[] args)
```

Comment:
This method isn't used in production code and has no tests. Should we remove this code until it is needed?